File: PHP/old_approach/old_approach.php
<?php
	function reverse_number(int $num) : int
	{
		$digit = 0;
		$sum = 0;

		while ($num != 0)
		{
			$digit = $num % 10;
			$sum = $sum * 10 + $digit;
			$num = intdiv($num, 10);
		}

		return $sum;
	}

	// 1234567890987654321
	$num = readline("Please insert a number: ");
	$res = reverse_number($num);
	echo $res;
?>

File: Nim/old_approach/old_approach.nim
import strutils

proc reverseNumber(number : int) : int =
  var digit : int
  var sum : int = 0
  var num : int = number

  while (num != 0):
    digit = num mod 10
    sum = sum * 10 + digit
    num = int(num / 10)

  return sum

var number : int = parseInt(readLine(stdin))
var res : int = reverse_number(number)
echo res